File: windows/CodexFlowWindows/src/CodexFlowWindows/Data/Models/DashboardModels.cs
using System.Text.Json.Serialization;

namespace CodexFlowWindows.Data.Models;

public sealed record DashboardResponse
{
    [JsonPropertyName("agent")]
    public AgentSnapshot Agent { get; init; } = new();

    [JsonPropertyName("stats")]
    public DashboardStats Stats { get; init; } = new();

    [JsonPropertyName("sessions")]
    public IReadOnlyList<SessionSummary> Sessions { get; init; } = Array.Empty<SessionSummary>();

    [JsonPropertyName("approvals")]
    public IReadOnlyList<PendingRequestView> Approvals { get; init; } = Array.Empty<PendingRequestView>();

    public static DashboardResponse Empty { get; } = new();
}

public sealed record AgentSnapshot
{
    [JsonPropertyName("connected")]
    public bool Connected { get; init; }

    [JsonPropertyName("startedAt")]
    public string StartedAt { get; init; } = string.Empty;

    [JsonPropertyName("listenAddr")]
    public string ListenAddr { get; init; } = string.Empty;

    [JsonPropertyName("codexBinaryPath")]
    public string CodexBinaryPath { get; init; } = string.Empty;
}

public sealed record DashboardStats
{
    [JsonPropertyName("totalSessions")]
    public int TotalSessions { get; init; }

    [JsonPropertyName("loadedSessions")]
    public int LoadedSessions { get; init; }

    [JsonPropertyName("activeSessions")]
    public int ActiveSessions { get; init; }

    [JsonPropertyName("pendingApprovals")]
    public int PendingApprovals { get; init; }
}

public sealed record HealthResponse
{
    [JsonPropertyName("ok")]
    public bool Ok { get; init; }

    [JsonPropertyName("timestamp")]
    public string Timestamp { get; init; } = string.Empty;
}

public sealed record OkResponse
{
    [JsonPropertyName("ok")]
    public bool Ok { get; init; }
}

File: android/CodexFlowAndroid/app/src/main/java/com/codexflow/codexflow/data/model/DashboardModels.kt
package com.codexflow.codexflow.data.model

import kotlinx.serialization.Serializable

@Serializable
data class DashboardResponse(
    val agent: AgentSnapshot = AgentSnapshot(),
    val stats: DashboardStats = DashboardStats(),
    val sessions: List<SessionSummary> = emptyList(),
    val approvals: List<PendingRequestView> = emptyList()
)

@Serializable
data class AgentSnapshot(
    val connected: Boolean = false,
    val startedAt: String = "",
    val listenAddr: String = "",
    val codexBinaryPath: String = ""
)

@Serializable
data class DashboardStats(
    val totalSessions: Int = 0,
    val loadedSessions: Int = 0,
    val activeSessions: Int = 0,
    val pendingApprovals: Int = 0
)

@Serializable
data class HealthResponse(
    val ok: Boolean = false,
    val timestamp: String = ""
)
